packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/notifications/BacklinksNotificationsTests.kt

```diff
@@ -408,4 +409,176 @@ class BacklinksNotificationsTests : RealmEntityNotificationTests {
             c.close()
         }
     }
+
+    @Test
+    override fun keyPath_topLevelProperty() = runBlocking<Unit> {
+        val c = Channel<ResultsChange<Sample>>(1)
+        realm.write {
+            copyToRealm(
+                Sample().apply {
+                    this.stringField = "parent"
+                    this.nullableObject = Sample().apply {
+                        this.stringField = "child"
+                    }
+                }
+            )
+        }
+        val result: RealmResults<Sample> = realm.query<Sample>("stringField = 'child'").find()
+        assertEquals(1, result.first().objectBacklinks.size)
+        val observer = async {
+            result.asFlow(listOf("objectBacklinks")).collect {
+                c.trySend(it)
+            }
+        }
+        assertIs<InitialResults<Sample>>(c.receiveOrFail())
+        realm.write {
+            val child = findLatest(result.first())!!
+            assertEquals("child", child.stringField)
+            copyToRealm(
+                Sample().apply {
+                    this.stringField = "newParent"
+                    this.nullableObject = child
+                }
+            )
+        }
+        c.receiveOrFail().let { resultsChange ->
+            assertIs<UpdatedResults<Sample>>(resultsChange)
+            when (resultsChange) {
+                is UpdatedResults -> {
+                    assertEquals(0, resultsChange.insertions.size)
+                    assertEquals(1, resultsChange.changes.size)
+                    assertEquals(0, resultsChange.deletions.size)
+                    assertEquals(2, resultsChange.list.first().objectBacklinks.size)
+                }
+                else -> fail("Unexpected change: $resultsChange")
+            }
+        }
+        observer.cancel()
+        c.close()
+    }
+
+    @Test
+    override fun keyPath_nestedProperty() = runBlocking<Unit> {
+        val c = Channel<ResultsChange<Sample>>(1)
+        realm.write {
+            copyToRealm(
+                Sample().apply {
+                    this.stringField = "parent"
+                    this.nullableObject = Sample().apply {
+                        this.stringField = "child"
+                    }
+                }
+            )
+        }
+        val result: RealmResults<Sample> = realm.query<Sample>("stringField = 'child'").find()
+        assertEquals(1, result.first().objectBacklinks.size)
+        val observer = async {
+            result.asFlow(listOf("objectBacklinks.intField")).collect {
+                c.trySend(it)
+            }
+        }
+        assertIs<InitialResults<Sample>>(c.receiveOrFail())
+        realm.write {
+            // Update field that should not trigger a notification
+            findLatest(result.first())!!.booleanField = false
+        }
```

Comment:
We should somehow assert that nothing happens. The channel result will be dropped on new events since the channel only have size 1, so theoretically this could emit a notification but the test could still succeed.

Reply:
Good point, I added an assert catching this case.